.github/workflows/expected-release-version.yml

```diff
@@ -0,0 +1,102 @@
+# add comment on PR for expected release version based on the commit messages
+
+name: Expected Release Version
+on:
+  - workflow_call
+
+jobs:
+  release-type:
+    name: Check Release Type
+    runs-on: ubuntu-latest
+    outputs:
+      release_level: ${{ steps.release_type.outputs.RELEASE_TYPE }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ env.GITHUB_SHA }}
+      - name: Get Commit Messages
+        id: get_commit_messages
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          COMMITS=$(curl -s -H "Accept: application/vnd.github.v3+json" \
+                          -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+                          "https://api.github.com/repos/${{ github.repository }}/pulls/$PR_NUMBER/commits")
+          COMBINED_MESSAGES=$(echo "$COMMITS" | jq -r '.[].commit.message' | paste -sd ' ')
+          echo "COMBINED_MESSAGES=$COMBINED_MESSAGES" >> $GITHUB_OUTPUT
+      - name: Get Release Type
+        id: release_type
+        run: |
+          if echo "${{ steps.get_commit_messages.outputs.COMBINED_MESSAGES }}" | grep -q "BREAKING CHANGE"; then
+            echo "RELEASE_TYPE=major" >> $GITHUB_OUTPUT
+          elif echo "${{ steps.get_commit_messages.outputs.COMBINED_MESSAGES }}" | grep -q "feat"; then
+            echo "RELEASE_TYPE=minor" >> $GITHUB_OUTPUT
+          elif echo "${{ steps.get_commit_messages.outputs.COMBINED_MESSAGES }}" | grep -q "fix"; then
+            echo "RELEASE_TYPE=patch" >> $GITHUB_OUTPUT
+          else
+            echo "RELEASE_TYPE=none" >> $GITHUB_OUTPUT
+          fi
+  determine-next-version:
+    name: Determine Next Version
+    needs: [release-type]
+    if: needs.release-type.outputs.release_level != 'none'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ env.GITHUB_SHA }}
+      - name: Get latest tag
+        uses: actions-ecosystem/action-get-latest-tag@v1
+        id: get-latest-tag
+      - name: Determine Next Version
+        id: next_version
+        run: |
+          echo ${{ steps.get-latest-tag.outputs.tag }}
+          IFS='v' read -r -a VERSION <<< "${{ steps.get-latest-tag.outputs.tag }}"
+          IFS='.' read -r -a VERSION_PARTS <<< "${VERSION[1]}"
```

Comment:
We should also test this in case when tag is created as only `major.minor.patch` instead of `v{tag}` in which case `VERSION[1]` won't have any data in it.